baremetal/metal3machine_manager_test.go

```diff
@@ -2302,6 +2302,57 @@ var _ = Describe("Metal3Machine manager", func() {
 		}),
 	)
 
+	DescribeTable("Test updateObj function, original object shall remain unchanged",
+		func(Host *bmh.BareMetalHost) {
+			c := fakeclient.NewFakeClientWithScheme(setupSchemeMm(), Host)
+			machineMgr, err := NewMachineManager(c, nil, nil, nil, nil, klogr.New())
+
+			Expect(err).NotTo(HaveOccurred())
+
+			tcHost := Host.DeepCopyObject()
+			err = machineMgr.updateObject(context.TODO(), Host)
+
+			Expect(err).NotTo(HaveOccurred())
+			Expect(tcHost).To(Equal(Host))
```

Comment:
Shouldn't you get the host object from the fakeclient rather than making a Deepcopy and comparing against it? How are you sure that updateObject() function has not changed the host???

Reply:
My goal is to compare the entire original host object '**tcHost**' to the host object '**Host**' given to the updateObj function. With DeepCopyObj I make sure that the entire object is copied to 'tcHost'.
In this case a simple assignment would likely work but deepcopy handles possible lists, maps etc correctly without user attention.

Reply:
My point is if the goal is to check the behavior of updateObj function, then shouldn't the object be fetched from the client and see how if the object has bee changed?

Reply:
My understanding is that this is achieved via the machineMgr.
Please correct me if I'm wrong.

Reply:
If you go through the machine manager test file, you will see in few other test cases the hosts are being fetched through the client after a function call (i.e. https://github.com/metal3-io/cluster-api-provider-metal3/blob/master/baremetal/metal3machine_manager_test.go#L632 ). That is the proper way to compare the changes in objects before and after an operation IMO.

Reply:
Thanks for the example. I got it, done.